### PR TITLE
Normalize the version number of the compiler

### PR DESCRIPTION
**Pull request by shindere**
Make sure it is 5.0.0 (major.minor.release) with the minor numbe
being just 0 rather than 00.

### CODE REVIEW

**Comment by shindere** [comment]
I thought I once saw an issue where this was mentionned, I am not
able to find it any longer.

If anybody knows, it'd be great to associate it to this PR.

**Comment by shindere** [comment]
Also: at the moment the PR is minimal. Should it go a bit further, e.g.
the manual says:

```
In earlier versions of OCaml, it was possible to use word-aligned
pointers to addresses outside the heap as OCaml values, just by
casting the pointer to type "value".  Starting with OCaml 4.11, this
usage is deprecated and will stop being supported in OCaml 5.0.0.
```

Shoould we for instance say something like:

```
After having been deprecated since OCaml 4.11, this possibility is finally
no longer supported from OCaml5.0.0 onwards.
```

**Comment by dbuenzli** [comment]
> I thought I once saw an issue where this was mentionned, I am not
> able to find it any longer.

It was https://github.com/ocaml/ocaml/pull/10757. The conclusion that were was no compelling reason to change it.

**Comment by shindere** [comment]
Daniel Bünzli (2022/02/22 23:23 -0800):
> > I thought I once saw an issue where this was mentionned, I am not
> > able to find it any longer.
>
> It was https://github.com/ocaml/ocaml/pull/10757. The conclusion that
> were was no compelling reason to change it.

Thank you so much!

Since then, the topic was discussed in a developer meeting and we agreed
to change the version number and this PR comes as a consequence of this
decision.

**Comment by dra27** [comment]
Thanks, @shindere! This needs FlexDLL 0.41 to be released (which I was planning on doing today), and then the submodule needs updating.

There are already some inconsistencies, but in the docs/comments I would expect `@since 5.00` to become `@since 5.0` (i.e. 5.0 rather than 5.0.0).

**Comment by dra27** [comment]
The change also needs a bootstrap to update the version reported by boot/ocamlc

**Comment by dra27** [comment]
> The change also needs a bootstrap to update the version reported by boot/ocamlc

Well, not exactly need, but it would be sensible for it to be in sync! Now I’ll stop replying to myself…

**Comment by shindere** [comment]
David Allsopp (2022/02/23 00:16 -0800):
> > The change also needs a bootstrap to update the version reported by boot/ocamlc
>
> Well, not exactly need, but it would be sensible for it to be in sync!
> Now I’ll stop replying to myself…

I was wondering yes, if I should bootstrap or not and thought that it
would be better to "save" a bootstrap and to wait until there are other
things one would want to have in. But I can definitely perform a
bootstrap if so desired.

**Comment by shindere** [comment]
David Allsopp (2022/02/23 00:13 -0800):
> Thanks, @shindere! This needs FlexDLL 0.41 to be released (which I was
> planning on doing today), and then the submodule needs updating.

Okay this can wait, no problem. Did you mean htat the submodule update
should be included in this PR? If so I can definitely add it, once
flexdll has been released.

> There are already some inconsistencies, but in the docs/comments I
> would expect ***@***.*** 5.00` to become ***@***.*** 5.0` (i.e. 5.0 rather
> than 5.0.0).

I was wondering, actually. Would be happy to hear others on this.
Shouldn't we always use full version numbers to remove any risk of
ambiguity?

**Comment by Octachron** [comment]
We don't add new functions (anymore) in patch-level releases. It thus seem better to not add this irrelevant patch-level version in `@since` tag.

**Comment by shindere** [comment]
Florian Angeletti (2022/02/23 00:46 -0800):
> We don't add new functions (anymore) in patch-level releases. It thus
> seem better to not add this irrelevant patch-level version in ***@***.***`
> tag.

Somehow what you write makes sense tome but at the same time, I am wondering:

 1. Are users aware of this? If they are not (which is what I assume is closer to the truth) then yes, incomplete version numbers will be ambiguous.
 1. Even for us, I'd feel it simpler and more straightforward to have the convension that we always write full version numbers, rather than  having to remember that the way to write them is context-dependent.

And what if one day, for  some reason, we do have to add or change a
function between two releases of the same minor verison?

**Comment by dra27** [comment]
> And what if one day, for some reason, we do have to add or change a function between two releases of the same minor verison?

This isn't the minor version, it's the _revision_ - we may be stretching the definition of compatible for the "minor" case but it is _absolutely_ the case that no functions should be added in a revision (4.02.2 is our dirty secret...)

The discussion is possible meandering, though - places where 5.00 was written before instead of 5.00.0 should be 5.0 for now. Deciding whether or not we want to put full version numbers everywhere is orthogonal to this PR.

**Comment by dra27** [comment]
The flexdll submodule is updated, so I just cycled the PR to force GitHub to recompute the merge commit (TL;DR AppVeyor should hopefully pass now)

**Comment by shindere** [comment]
David Allsopp (2022/02/23 02:28 -0800):
> The flexdll submodule is updated, so I just cycled the PR to force
> GitHub to recompute the merge commit (TL;DR AppVeyor should hopefully
> pass now)

Thanks. I rebased on top of that.

**Comment by shindere** [comment]
David Allsopp (2022/02/23 02:28 -0800):
> > And what if one day, for some reason, we do have to add or change a function between two releases of the same minor verison?
>
> This isn't the minor version, it's the _revision_ - we may be
> stretching the definition of compatible for the "minor" case but it is
> _absolutely_ the case that no functions should be added in a revision
> (4.02.2 is our dirty secret...)

That's what I meant, just the wording was not clear enough, I guess.

> The discussion is possible meandering, though - places where 5.00 was
> written before instead of 5.00.0 should be 5.0 for now. Deciding
> whether or not we want to put full version numbers everywhere is
> orthogonal to this PR.

Fair enough. I took that into account and updated the PR accordingly.

File: manual/src/cmds/intf-c.etex [hunk 1]
```diff
@@ -464,7 +464,7 @@ floating-point numbers.}
 In earlier versions of OCaml, it was possible to use
 word-aligned pointers to addresses outside the heap as OCaml values,
 just by casting the pointer to type "value".  Starting with OCaml
-4.11, this usage is deprecated and will stop being supported in OCaml 5.00.
+4.11, this usage is deprecated and will stop being supported in OCaml 5.0.
```
Comment (xavierleroy) [inline manual/src/cmds/intf-c.etex]: You can be more assertive:

This usage is no longer supported since OCaml 5.0.

**Comment by shindere** [comment]
Xavier Leroy (2022/02/23 08:11 -0800):
> You can be more assertive:

Done, thanks. Changes entry added.

**Review by xavierleroy — APPROVED** [review]
LGTM.  Merging.

**Pull request merged**